wolfcrypt/test/test.c

```diff
@@ -29838,6 +29857,3977 @@ WOLFSSL_TEST_SUBROUTINE int ed448_test(void)
+        0x74, 0x40, 0x5B, 0x57, 0xE3, 0x48, 0x4A, 0x52
+    };
+    const byte kyber1024_ss[] = {
+        0xC1, 0x4E, 0xBD, 0x6E, 0x37, 0x88, 0xA6, 0x41,
+        0xD1, 0x75, 0x5B, 0x4C, 0x86, 0x9C, 0x46, 0x99,
+        0x4F, 0x75, 0xAF, 0x16, 0xB4, 0x0F, 0x8F, 0x64,
+        0x92, 0xCC, 0xBE, 0xC7, 0xA3, 0xDA, 0x9B, 0xFE
+    };
+#endif
+
+    ret = wc_KyberKey_Init(KYBER1024, &key, HEAP_HINT, INVALID_DEVID);
+    if (ret != 0)
+        return -20301;
+
+    ret = wc_KyberKey_MakeKeyWithRandom(&key, kyber1024_rand,
+        sizeof(kyber1024_rand));
+    if (ret != 0)
+        return -20302;
+
+    ret = wc_KyberKey_EncodePublicKey(&key, pub, sizeof(pub));
+    if (ret != 0)
+        return -20303;
+
+    ret = wc_KyberKey_EncodePrivateKey(&key, priv, sizeof(priv));
+    if (ret != 0)
+        return -20304;
+
+    if (XMEMCMP(pub, kyber1024_pk, sizeof(kyber1024_pk)) != 0)
+        return -20305;
+
+    if (XMEMCMP(priv, kyber1024_sk, sizeof(kyber1024_sk)) != 0)
+        return -20306;
+
+    ret = wc_KyberKey_EncapsulateWithRandom(&key, ct, ss, kyber1024enc_rand,
+        sizeof(kyber1024enc_rand));
+    if (ret != 0)
+        return -20307;
+
+    if (XMEMCMP(ct, kyber1024_ct, sizeof(kyber1024_ct)) != 0)
+        return -20308;
+
+    if (XMEMCMP(ss, kyber1024_ss, sizeof(kyber1024_ss)) != 0)
+        return -20309;
+
+    ret = wc_KyberKey_Decapsulate(&key, ss_dec, ct, sizeof(kyber1024_ct));
+    if (ret != 0)
+        return -20310;
+
+    if (XMEMCMP(ss_dec, kyber1024_ss, sizeof(kyber1024_ss)) != 0)
+        return -20311;
+
+    wc_KyberKey_Free(&key);
+
+    return 0;
+}
+#endif /* WOLFSSL_KYBER1024 */
+#endif /* WOLFSSL_WC_KYBER */
+
+WOLFSSL_TEST_SUBROUTINE int kyber_test(void)
+{
+    int ret;
+    KyberKey key;
+    WC_RNG rng;
+    int i;
+    byte priv[KYBER_MAX_PRIVATE_KEY_SIZE];
+    byte pub[KYBER_MAX_PUBLIC_KEY_SIZE];
+    byte priv2[KYBER_MAX_PRIVATE_KEY_SIZE];
+    byte pub2[KYBER_MAX_PUBLIC_KEY_SIZE];
+    byte ct[KYBER_MAX_CIPHER_TEXT_SIZE];
+    byte ss[KYBER_SS_SZ];
+    byte ss_dec[KYBER_SS_SZ];
+    int testData[][4] = {
+    #ifdef WOLFSSL_KYBER512
+        { KYBER512,  KYBER512_PRIVATE_KEY_SIZE,  KYBER512_PUBLIC_KEY_SIZE,
+          KYBER512_CIPHER_TEXT_SIZE },
+    #endif
+    #ifdef WOLFSSL_KYBER768
+        { KYBER768,  KYBER768_PRIVATE_KEY_SIZE,  KYBER768_PUBLIC_KEY_SIZE,
+          KYBER768_CIPHER_TEXT_SIZE },
+    #endif
+    #ifdef WOLFSSL_KYBER1024
+        { KYBER1024, KYBER1024_PRIVATE_KEY_SIZE, KYBER1024_PUBLIC_KEY_SIZE,
+          KYBER1024_CIPHER_TEXT_SIZE },
+    #endif
+    };
+
+#ifndef HAVE_FIPS
+    ret = wc_InitRng_ex(&rng, HEAP_HINT, INVALID_DEVID);
+#else
+    ret = wc_InitRng(&rng);
+#endif
+    if (ret != 0)
+        return -20000;
+
+    for (i = 0; i < (int)(sizeof(testData) / sizeof(*testData)); i++) {
+        ret = wc_KyberKey_Init(testData[i][0], &key, HEAP_HINT, INVALID_DEVID);
+        if (ret != 0)
+            return -20001 - i * 20;
+
+        ret = wc_KyberKey_MakeKey(&key, &rng);
```

Comment:
Tests are reporting some leaking memory, please give it a once over with -fsanitize=address on a linux like system

./configure --enable-kyber --with-liboqs=/usr/local CC="clang -fsanitize=address" --enable-debug
make
./wolfcrypt/test/testwolfcrypt

```

Direct leak of 80 byte(s) in 1 object(s) allocated from:
    #0 0x4ded50 in malloc (/home/jak/Documents/wolfssl/wolfcrypt/test/.libs/testwolfcrypt+0x4ded50)
    #1 0x7f73a8561dbd in OQS_KEM_kyber_512_new (/home/jak/Documents/wolfssl/src/.libs/libwolfssl.so.35+0x3f3dbd)
    #2 0x534296 in kyber_test /home/jak/Documents/wolfssl/wolfcrypt/test/test.c:33751:15
    #3 0x519073 in wolfcrypt_test /home/jak/Documents/wolfssl/wolfcrypt/test/test.c:1360:17
    #4 0x535b47 in main /home/jak/Documents/wolfssl/wolfcrypt/test/test.c:1619:9
    #5 0x7f73a71bdc86 in __libc_start_main /build/glibc-CVJwZb/glibc-2.27/csu/../csu/libc-start.c:310
```

Reply:
fixed